### DISCUSSION TITLE
FCM

### DISCUSSION THREAD

**Discussion by radlik01** [DelphiWorlds/Kastri#173] [Q&A]
Hello,

I have one question regarding FCM messaging. I would like to receive the FCM message like JSON and not like plain text but I do not know how to do it.
`procedure TSafeTrackerForm.FCMMessageReceivedHandler(Sender: TObject;
const APayload: TStrings);
begin
  TOSLog.d('TsafeTracker %s begin token %s', ['FCMMessageReceivedHandler', APayload.Text]);
  mmMessage.ClearContent;
  mmMessage.Lines.AddStrings(APayload);
end;`

Is it possible to get Payload as JSON ?

Thank you

Jan

**Comment by DelphiWorlds** [comment]
You could either use the newer and [recommended implementation here](https://github.com/DelphiWorlds/Kastri/tree/master/Demos/FCMRebooted) which offers the payload as JSON already, or you could use the inbuilt TJSONObject class (in the System.JSON unit) to parse the payload:

```
var
   LJSON: TJSONValue;
begin
  LJSON := TJSONObject.ParseJSONValue(APayload.Text);
  try
    // Do something with LJSON here
  finally
    LJSON.Free;
  end;
end;
```

**Comment by radlik01** [comment]
Thank you,

the problem is, that when I use the unit DW.FCMManager the aplication freez on startup with any message in log.

**Comment by Morrismx** [comment]
I have the same problem Mauricio HerreraMex Dial 5520959678US Dial 01152-5520959678El 2 jun 2023, a la(s) 4:20 a.m., radlik01 ***@***.***> escribió:﻿
Thank you,
the problem is, that when I use the unit DW.FCMManager the aplication freez on startup with any message in log.

**Comment by DelphiWorlds** [comment]
Are you compiling for 64-bit? See this issue: https://github.com/DelphiWorlds/Kastri/issues/174
